chemprop/v2/cli/train.py

```diff
@@ -32,41 +32,308 @@ class TrainSubcommand(Subcommand):
 
     @classmethod
     def add_args(cls, parser: ArgumentParser) -> ArgumentParser:
-        return add_args(parser)
+        parser = add_common_args(parser)
+        return add_train_args(parser)
 
     @classmethod
     def func(cls, args: Namespace):
         process_args(args)
+        validate_args(args)
         main(args)
 
+def add_common_args(parser: ArgumentParser) -> ArgumentParser:
+    parser.add_argument(
+        "--logdir",
+        nargs="?",
+        const="chemprop_logs",
+        help="runs will be logged to {logdir}/chemprop_{time}.log. If unspecified, will use 'save_dir'. If only the flag is given (i.e., '--logdir'), then will write to 'chemprop_logs'",
+    )
 
-def add_args(parser: ArgumentParser) -> ArgumentParser:
+    data_args = parser.add_argument_group("input data parsing args")
+    data_args.add_argument(
+        "-s",
+        "--smiles_columns",
+        type=list[str],
+        # to do: make sure default is coded correctly
+        help="List of names of the columns containing SMILES strings. By default, uses the first :code:`number_of_molecules` columns.",
+    )
+    data_args.add_argument(
+        "--number_of_molecules",
+        type=int,
+        default=1,
+        help="Number of molecules in each input to the model. This must equal the length of :code:`smiles_columns` (if not :code:`None`).",
+    )
+    data_args.add_argument(
+        "--checkpoint_dir",
+        type=str,
+        help="Directory from which to load model checkpoints (walks directory and ensembles all models that are found).",
+    )
+    data_args.add_argument(
+        "--checkpoint_path",
+        type=str,
+        help="Path to model checkpoint (:code:`.pt` file).",
+    )
+    data_args.add_argument(
+        "--checkpoint_paths",
+        type=list[str],
+        help="List of paths to model checkpoints (:code:`.pt` files).",
+    )
+    data_args.add_argument(
+        "--no_cuda",
+        action="store_true",
+        help="Turn off cuda (i.e., use CPU instead of GPU).",
+    )
+    data_args.add_argument(
+        "--gpu",
+        type=int,
+        help="Which GPU to use.",
+    )
```

Comment:
can be controlled external to chemprop

Reply:
I'm not sure how it can be externally controlled. I think we need these arguments so that the user can tell lightning if and which gpu to use. Is there another way to tell lightning what to do, maybe you mean environment variables?

Reply:
this can be controlled via `$CUDA_VISIBLE_DEVICES` environment variable